src/zcash/Note.cpp

```diff
@@ -188,6 +188,23 @@ std::optional<SaplingNote> SaplingNotePlaintext::note(const SaplingIncomingViewi
     }
 }
 
+std::pair<SaplingNotePlaintext, SaplingPaymentAddress> SaplingNotePlaintext::from_rust(
+    rust::Box<wallet::DecryptedSaplingOutput> decrypted)
+{
+    SaplingPaymentAddress pa(
+        decrypted->recipient_d(),
+        uint256::FromRawBytes(decrypted->recipient_pk_d()));
+    SaplingNote note(
+        pa.d,
+        pa.pk_d,
+        decrypted->note_value(),
+        uint256::FromRawBytes(decrypted->note_rseed()),
+        decrypted->zip_212_enabled() ? Zip212Enabled::AfterZip212 : Zip212Enabled::BeforeZip212);
```

Comment:
The [SaplingNote constructor](https://github.com/zcash/zcash/blob/1b7b73b7941ad4ddc73dbcc8d767a14138bf4b31/src/zcash/Note.cpp#L48-L63) sets `rseed` randomly if `zip212Enabled` is `AfterZip212`. Shouldn't it be setting it from the decrypted `rseed` field here?

Reply:
You're looking at the wrong constructor. The one used here is:
https://github.com/zcash/zcash/blob/1b7b73b7941ad4ddc73dbcc8d767a14138bf4b31/src/zcash/Note.hpp#L78-L79